SCons/Tool/__init__.py

```diff
@@ -251,7 +251,7 @@ def __call__(self, env, *args, **kw) -> None:
                 kw.update(call_kw)
             else:
                 kw = self.init_kw
-        env.Append(TOOLS=[self.name])
+        env.AppendUnique(TOOLS=[self.name])
```

Comment:
Is it possible that calling a tool different dict more than one time might be useful? And so appendunique would drop any 2-n'th tool initialization?

Reply:
Interesting thought. But... nothing's preventing things from being called more than once, `TOOLS` is just a list of strings appended when a tool is called, I'm not seeing any benefit to having that information recorded multiple times, as it won't indicate any differences in invocation (like different kwargs) anyway.  *Almost* nothing uses `TOOLS` for anything after that one init-check : if `tools` is `None` use `TOOLS`.  The msvc tool does use it for a check, that's the exception.

Reply:
ahh true. it's just saving the name, so it they were reinvoked any second or further all would not yield anything different with out some evil hackery..